### PR TITLE
Add helper to determine the atomic identifier for attribute Id given

### PR DESCRIPTION
**Pull request by brdandu**
- as_underlying_atomic_identifier_for_attributeId checks for the attribute type and returns the atomic identifier associated with it.
- JIRA: ZAPP-1385

### CODE REVIEW

**Review by tecimovic — APPROVED** [review]

File: src-electron/generator/helper-attribute.js [hunk 1]
```diff
@@ -76,6 +79,89 @@ async function attributeDefault(options) {
...
+    : null
+  // If attribute type directly points to the atomic type then return that
+  if (atomicInfo) {
+    // All types in the number and string table should be found here.
+    return atomicInfo.atomicId
+  } else {
+    // If attribute type does not point to atomic type
+    let dataType = await queryZcl.selectDataTypeByNameAndClusterId(
+      this.global.db,
+      attributeDetails.type,
```
Comment (tecimovic) [inline src-electron/generator/helper-attribute.js:109]: This whole block of code is a bit weird.... Mining in all these different places for a name: I can understand where it comes from, but shouldn't there be a better way to do this? Just one table "TYPES" that contains all known types, be it enum values or bitmap values or whatever?

I wonder if having separate tables for these was a mistake...

Comment (brdandu) [reply src-electron/generator/helper-attribute.js:109]: All types are in the data_type table but that only has the common characteristics with the distinct attributes of each type going into their corresponding tables.

File: src-electron/generator/helper-attribute.js [hunk 2]
```diff
@@ -76,6 +79,89 @@ async function attributeDefault(options) {
   return templateUtil.templatePromise(this.global, p)
 }
 
+/**
+ * Given an attribute Id determine its corresponding atomic identifier from the
+ * atomic table.
+ * @param {*} attributeId
+ */
+async function as_underlying_atomic_identifier_for_attribute_id(attributeId) {
```
Comment (dhchandw) [inline src-electron/generator/helper-attribute.js:87]: I might be wrong but doesn't matter have derived types from integers and floats. What would happen in that case?

Comment (brdandu) [reply src-electron/generator/helper-attribute.js:87]: Don't know of any derived floats, ints or strings. Could you point me to any you ran across them?
if that is the case then need to add additional logic to handle that.

Comment (dhchandw) [reply src-electron/generator/helper-attribute.js:87]: does selectAtomicType() take care of that?

Comment (brdandu) [reply src-electron/generator/helper-attribute.js:87]: I don't think we have any other tags apart from enum, bitmap and struct so I doubt it.

Comment (dhchandw) [reply src-electron/generator/helper-attribute.js:87]: Okay I'm getting confused here, we don't NEED to return baseline atomic types here right?

Comment (dhchandw) [reply src-electron/generator/helper-attribute.js:87]: So returning a derived atomic data type like clusterId would be acceptable here?

Comment (brdandu) [reply src-electron/generator/helper-attribute.js:87]: does selectAtomicType() take care of that? => Yes that takes care of strings, ints, floats

Comment (brdandu) [reply src-electron/generator/helper-attribute.js:87]: This has nothing to do with base types. It is about returning the Atomic_identfier from the atomic table based on attribute type

Comment (dhchandw) [reply src-electron/generator/helper-attribute.js:87]: Got it, sorry for the confusion.

**Review by dhchandw — APPROVED** [review]

**Pull request merged**